pkg/nsx/services/ipaddressallocation/ipaddressallocation.go

```diff
@@ -180,6 +183,67 @@ func (service *IPAddressAllocationService) DeleteIPAddressAllocation(obj interfa
 	return nil
 }
 
+func (service *IPAddressAllocationService) DeleteIPAddressAllocationByNamespacedName(namespace, name string) error {
+	// First, we need to find the IPAddressAllocation in our store
+	allIPAddressAllocations := service.ipAddressAllocationStore.List()
+	var targetIPAddressAllocation *model.VpcIpAddressAllocation
+
+	for _, obj := range allIPAddressAllocations {
+		ipAddressAllocation, ok := obj.(*model.VpcIpAddressAllocation)
+		if !ok {
+			continue
+		}
+
+		for _, tag := range ipAddressAllocation.Tags {
+			if *tag.Scope == common.TagScopeNamespace && *tag.Tag == namespace {
+				targetIPAddressAllocation = ipAddressAllocation
+				break
+			}
+		}
+
+	}
+
+	if targetIPAddressAllocation == nil {
+		log.Error(nil, "IPAddressAllocation with namespace %s and name %s not found", namespace, name)
+		return nil
+	}
+
+	// To prevent the situation where a resource is not found when entering the reconcile phase,
+	// but a same name CR is created by the time the delete operation is performed
+	uid := service.GetIPAddressAllocationUID(targetIPAddressAllocation)
+	uids, err := service.GetAllIPAddressAllocationUIDs(namespace)
+	if err != nil {
+		log.Error(err, "Failed to get all IPAddressAllocation UIDs in namespace %s", namespace)
+		return err
+	}
+	if uids.Has(uid) {
+		log.Error(nil, "The same IPAddressAllocation with namespace %s and name %s already exists", namespace, name)
+		return nil
```

Comment:
As the comments above, we may skip the case here instead of returning

Reply:
ok